### PR TITLE
Angularify badge component.

### PR DESCRIPTION
**Pull request by k0nG**
TP Story:
https://rockabox.tpondemand.com/entity/8475

Task:
https://rockabox.tpondemand.com/entity/8284

Angularised/Angularfied badge component. Depends on https://github.com/rockabox/rbx_style_guide/pull/151

### CODE REVIEW

File: src/rb-badge/rb-badge-directive.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        // Set body text for warning state. Defaults to &nbsp; in template
+        if (angular.isDefined(attr.body) && attr.state == 'warning') {
+            scope.body = attr.body + ' !';
+        }
+
+        // Set is-collapsed class
+        if (angular.isDefined(attr.collapsed)) {
+            elem.addClass('is-collapsed');
+        }
```
Comment (chrisvxd) [inline src/rb-badge/rb-badge-directive.js]: I prefer use of ng-class over assigning classes in js

Comment (k0nG) [inline src/rb-badge/rb-badge-directive.js]: No that makes sense. So pass it through on the scope then use ng-if. I'll update.

Comment (nwhite89) [inline src/rb-badge/rb-badge-directive.js]: should be able to do
```
ng-class="{
  'is-collapsed': collapsed
}"
```

take a look here https://github.com/rockabox/rb-table/blob/v1/src/directives/table/table.tpl.html#L19

File: src/rb-badge/rb-badge-directive.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     *    <rb-badge state="warning" body="6!"></rb-badge>
+     *    <rb-badge state="statusComplete"></rb-badge>
+     * </hljs>
+     *
+     * @ngInject
+     */
+    function rbBadgeDirective () {
+
+        return {
+            scope: {},
```
Comment (chrisvxd) [inline src/rb-badge/rb-badge-directive.js]: You should use isolated scope for the attrs so you can stop handling vars via `attr` in `link`

File: src/rb-badge/rb-badge-directive.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+        // Set body text for warning state. Defaults to &nbsp; in template
+        if (angular.isDefined(attr.body) && attr.state == 'warning') {
+            scope.body = attr.body + ' !';
+        }
+
+        // Set is-collapsed class
+        if (angular.isDefined(attr.collapsed)) {
+            scope.collapsed = true;
+        }
```
Comment (chrisvxd) [inline src/rb-badge/rb-badge-directive.js]: This method will go if we use isolated scopes

**Pull request merged**